airflow/models/mappedoperator.py

```diff
@@ -278,6 +278,11 @@ def __attrs_post_init__(self):
         for k, v in self.partial_kwargs.items():
             if k in self.template_fields:
                 XComArg.apply_upstream_relationship(self, v)
+        if self.partial_kwargs.get('sla'):
+            raise AirflowException(
+                f"SLAs are unsupported with mapped tasks. Please set `sla=None` for task "
+                f"{self.task_id!r}."
+            )
```

Comment:
Would it be cleaner to do this in `partial()` in `baseoperator.py` instead?

Reply:
sure can do

Reply:
i am only scratching the surface with this code so i appreciate the hand holding

Reply:
Hmm

```pycon
>>> import datetime
>>> bool(datetime.timedelta())
False
```

so we should do `is not None` here instead to not accidentally allowing a zero-duration timedelta.

Reply:
yeah true

Reply:
Should we consider chaning it to a notice/warnning and set it to None rather than raising exception?